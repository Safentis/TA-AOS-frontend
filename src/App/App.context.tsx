import { createContext, useContext } from 'react';
import { GlobalContent } from './App.interface';

export const GlobalContext = createContext<GlobalContent>({
  entries: [],
  numberOfPages: 0,
  numberOfItems: 0,
  currentPage: 1,
  handlePagination: () => {},
  filter: {
    status: "",
    gender: "",
    search: "",
  },
  setFilter: () => {}
});

export const useGlobalContext = () => useContext(GlobalContext);
