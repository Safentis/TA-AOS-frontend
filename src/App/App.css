.application {
  height: 100vh;
}

.application__inner {
  display: grid;
  grid-template-columns: 0.1fr 1fr;
  grid-template-rows: 0.1fr 1fr;
  grid-template-areas:
    'a h h h'
    'a m m m'
    'a m m m';
  height: 100%;
}

.application__header {
  grid-area: h;
}

.application__aside {
  grid-area: a;
}

.application__main {
  grid-area: m;
}

@media screen and (max-width: 1024px) {
  .application__inner {
    grid-template-rows: 0.15fr 0.25fr 1fr;
    grid-template-areas:
      'a a a'
      'h h h'
      'm m m';
  }
}
