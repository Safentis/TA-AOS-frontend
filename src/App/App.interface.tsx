export interface Props {}
export interface Entry {
  created: string;
  episode: string[];
  gender: string;
  id: number;
  image: string;
  location: {
    name: string;
    url: string;
  };
  name: string;
  origin: {
    name: string;
    url: string;
  };
  species: string;
  status: string;
  type: string;
  url: string;
}
export interface GlobalContent {
  entries: Entry[];
  numberOfPages: number;
  numberOfItems: number;
  currentPage: number;
  filter: Filter;
  handlePagination: (page: number) => void;
  setFilter: React.Dispatch<React.SetStateAction<Filter>>;
}

export interface Filter {
  status: string;
  gender: string;
  search: string;
}
