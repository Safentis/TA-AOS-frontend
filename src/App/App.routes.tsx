import React from 'react';
import { Switch, Route } from 'react-router-dom';

import { routes } from '../routes/routes';
import { RouteObject } from '../routes/routes.interface';

export default function MainRoutes(): React.ReactElement {
  return (
    <Switch>
      {routes.map(
        ({ path, component }: RouteObject, index: number) => (
          <Route
            path={path}
            component={component}
            exact={true}
            key={index}
          />
        ),
      )}
    </Switch>
  );
}
