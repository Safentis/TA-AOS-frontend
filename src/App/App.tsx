import React, { FC } from 'react';

import Header from '../layouts/Header';
import Aside from '../layouts/Aside';
import Main from '../layouts/Main';
import MainRoutes from './App.routes';

import './App.css';
import { Props } from './App.interface';

const App: FC<Props> = ({}): React.ReactElement => {
  return (
    <div className="application">
      <div className="application__inner">
        <Header className="application__header" />
        <Aside className="application__aside" />
        <Main className="application__main">
          <MainRoutes />
        </Main>
      </div>
    </div>
  );
};

export default App;
