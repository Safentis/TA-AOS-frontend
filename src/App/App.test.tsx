import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import App from './App';

import { Props } from './App.interface';

const setUp = (props: Props) => {
  return shallow(<App {...props} />);
};

describe('<App />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {};

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 Header element', () => {
      expect(component.find('Header')).toBeTruthy();
    });

    it('has 1 Aside element', () => {
      expect(component.find('Aside')).toBeTruthy();
    });

    it('has 1 Main element', () => {
      expect(component.find('Main')).toBeTruthy();
    });

    it('has 1 AppRoutes element', () => {
      expect(component.find('AppRoutes')).toBeTruthy();
    });
  });
});
