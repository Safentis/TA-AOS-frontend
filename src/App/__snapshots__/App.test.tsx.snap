// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component does match to the snapshot 1`] = `
<div
  className="application"
>
  <div
    className="application__inner"
  >
    <Header
      className="application__header"
    />
    <Aside
      className="application__aside"
    />
    <Main
      className="application__main"
    >
      <MainRoutes />
    </Main>
  </div>
</div>
`;
